src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -92,113 +93,116 @@ public FilingsService(
   }
 
   public FilingApi generateOverseasEntityFiling(
-      String requestId,
-      String overseasEntityId,
-      Transaction transaction,
-      String passThroughTokenHeader)
-      throws SubmissionNotFoundException, ServiceException {
+          String requestId,
+          String overseasEntityId,
+          Transaction transaction,
+          String passThroughTokenHeader)
+          throws SubmissionNotFoundException, ServiceException {
     var filing = new FilingApi();
     filing.setKind(FILING_KIND_OVERSEAS_ENTITY);
     setFilingApiData(filing, requestId, overseasEntityId, transaction, passThroughTokenHeader);
     return filing;
   }
 
   private void setFilingApiData(
-      FilingApi filing,
-      String requestId,
-      String overseasEntityId,
-      Transaction transaction,
-      String passThroughTokenHeader)
-      throws SubmissionNotFoundException, ServiceException {
+          FilingApi filing,
+          String requestId,
+          String overseasEntityId,
+          Transaction transaction,
+          String passThroughTokenHeader)
+          throws SubmissionNotFoundException, ServiceException {
     var logMap = new HashMap<String, Object>();
     logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
     logMap.put(TRANSACTION_ID_KEY, transaction.getId());
 
     Map<String, Object> userSubmission = new HashMap<>();
 
+//    OverseasEntitySubmissionDto submissionDto = setSubmissionData(userSubmission, overseasEntityId, logMap);
     OverseasEntitySubmissionDto submissionDto =
-        setSubmissionData(userSubmission, overseasEntityId, logMap);
-    setPaymentData(userSubmission, transaction, passThroughTokenHeader, logMap);
+            overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId)
+                    .orElseThrow(() -> new SubmissionNotFoundException(
+                            String.format("Empty submission returned when generating filing for %s", overseasEntityId)
+                    ));
 
     if (submissionDto.isForUpdate()) {
-      publicDataRetrievalService.initialisePublicData(
-          submissionDto.getEntityNumber(), passThroughTokenHeader);
+      publicDataRetrievalService.initialisePublicData(submissionDto.getEntityNumber(), passThroughTokenHeader);
       privateDataRetrievalService.initialisePrivateData(submissionDto.getEntityNumber());
+      UpdateSubmission updateSubmission = new UpdateSubmission();
+      setUpdateSubmissionData(updateSubmission, submissionDto, logMap);
+      Map<String, Object> updateSubmissionMap = objectMapper.convertValue(updateSubmission, new TypeReference<Map<String, Object>>() {});
```

Comment:
userSubmission - now used only for registration is just a new HashMap(). So why is updateSubmissioMap using an objectMapper? Can't we just use updateSubmission for both update and registration setData()?

Reply:
It won't let me put updateSubmission into the data Object because they are mismatching types.
![image](https://github.com/companieshouse/overseas-entities-api/assets/124156289/927d427d-7e09-496d-acf0-3fbccd46d5d2)

Reply:
But everything is an object!

Reply:
> But everything is an Object!

Well that is true. I just got it into my head that I wanted to use matching objects, lol. Okay I'll go with the userSubmission Map for Updates also, to avoid refactoring any current code for Registration.